### DISCUSSION TITLE
How to add syntax highlighting for a custom language?

### DISCUSSION THREAD

**Discussion by Andrew15-5** [typst/typst#2018] [General]
I've seen #768. Now Typst uses `syntect` which doesn't add new languages for now. I also have seen somewhere that custom language can be added with `#show regex` or something like that, but this isn't flexible enough. You can use custom theme files right in Typst, but you can't add a new language definition. I also read that `bat` is an alternative (?) that has much more languages, but Typst doesn't use it (and it's a CLI program only).

What are the options for adding new language that can be highlighted in `#raw`? Do I need to customize some rust package and then rebuild Typst? Maybe something harder/easier? Please tell.

I want at least be able to fiddle and be able to have a usable `typst` binary in my toolbox (that supports my custom language). In other words, I don't want to create an official language definition (or ask someone to do it) or anything like that. Just something for myself.

**Comment by laurmaedje** [comment]
You can load sublime syntaxes: https://typst.app/docs/reference/text/raw/#parameters-syntaxes

**Reply by Andrew15-5** [reply]
Oh, my bad. I'll check it out. Thanks.
